pkg/metabase/grant.go

```diff
@@ -0,0 +1,397 @@
+package metabase
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"errors"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/navikt/nada-backend/pkg/graph/models"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	iam "google.golang.org/api/iam/v1"
+)
+
+func (m *Metabase) grantMetabaseAccess(ctx context.Context, dsID uuid.UUID, subject string) {
```

Comment:
Is there a special reaon for implementing metabase access control with an event driven model? From my point of view, It cannot return errors to the dataset access control, leaving potential inconsistency between metabase and our database, and the user does not even know if there is an error with request to metabase during granting access of a dataset

Reply:
I definitely see your point on how it cannot return any errors. I think the main reason was that we wanted to separate the metabase part from the rest of the access handling so that it does not affect the bigquery access grants for users/groups if something is wrong with the metabase integration. The plan was to eventually have an external message bus (e.g. kafka/pubsub) that nada-backend writes to and a separate service entirely for handling everything to do with the metabase access.

It's a log time ago that this was decided, and we could revisit it, however I think that should be a separate task not included in this PR.